servicemesh-install/topics/appendix-threescale-adapter.adoc

```diff
@@ -96,86 +106,108 @@ The adapter includes a tool that allows you to generate the `handler`, `instance
 
-If you need to support multiple services, an additional label is required. It should be unique per service, for example `service-mesh.3scale.net/uid` as a default. You can to modify sample templates as required.
+Integration of a service requires that the above label be added to PodTemplateSpec on the Deployment of the target workload. The value, `threescale`, refers to the name of the generated handler. This handler will store the access token required to call 3scale.
 
-You must add a label to the `PodTemplateSpec` on the Deployment of the target workload. For example, in a `productpage` service, whose Pod is managed by the `productpage-v1` deployment, add the above label under `spec.template.labels` in `productpage-v1`, to have the adapter authorize requests to this service.
+Add the following label to the workload to pass the service ID to the adapter via the instance at request time:
+
+----
+destination.labels["service-mesh.3scale.net/service-id"] == "replace-me"
+----
 
+Your 3scale administrator should be able to provide you with both the required credentials name and the service ID.
 
+[[threescale-configure-integration]]
 = Configure the integration settings in 3scale
 
 [NOTE]
 ====
 For 3scale SaaS customers, {ProductName} is enabled as part of the Early Access program.
 ====
 
+[[threescale-integration-settings]]
 == Integration settings
 . Navigate to *[your_API_name] > Integration > Configuration*.
 . At the top of the *Integration* page click on *edit integration settings* in the top right corner.
 . Under the *Service Mesh* heading, click the *Istio* option.
 . Scroll to the bottom of the page and click *Update Service*.
 
+[[threescale-caching]]
 = Caching behavior
 Responses from 3scale System API's are cached by default within the adapter. Entries will be purged from the cache when they become older than the `cacheTTLSeconds` value. Also by default, automatic refreshing of cached entries will be attempted seconds before they expire, based on the `cacheRefreshSeconds` value. You can disable automatic refreshing by setting this value higher than the `cacheTTLSeconds` value.
 
 Caching can be disabled entirely by setting `cacheEntriesMax` to a non-positive value.
 
 By using the refreshing process, cached values whose hosts become unreachable will be retried before eventually being purged when past their expiry.
 
+[[threescale-authenticating-requests]]
 = Authenticating requests
 This Technology Preview release supports the following authentication methods:
 
 * *Standard API Keys*: single randomized strings or hashes acting as an identifier and a secret token.
 * *Application identifier and key pairs*: immutable identifier and mutable secret key strings.
 * *OpenID authentication method*: client ID string parsed from the JSON Web Token.
```

Comment:
It might be helpful to include the abbreviations here (API Key, AppID, OIDC) to help match this to the custom resource definition.

Reply:
I explained AppID and OIDC, but 3Scale _is_ an API Management product.